### PR TITLE
added bar chart

### PR DESCRIPTION
**Pull request by rohitgupta274509**

### CODE REVIEW

File: application/backend/src/main/resources/apps/typerefinery/components/structure/page/body-end.html [hunk 1]
```diff
@@ -89,8 +90,12 @@
     vueApp.component("p-column", primevue.column);
 
   })();
-  
-
```
Comment (sanjayaapna) [inline application/backend/src/main/resources/apps/typerefinery/components/structure/page/body-end.html:93]: plz, remove commented code.

Comment (rohitgupta274509) [reply application/backend/src/main/resources/apps/typerefinery/components/structure/page/body-end.html:93]: commented code removed.

File: application/backend/src/main/resources/apps/typerefinery/components/content/barchart/barchart.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<sly data-sly-use.model="io.typerefinery.websight.models.components.BarChart"
+     data-sly-use.template="./template.html"
+     data-sly-call="${template.barchart @ model=model}">
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/content/barchart/barchart.html:3]: Please follow the camel case naming convention.

barchart to barChart

File: application/backend/src/main/resources/apps/typerefinery/components/content/barchart/script.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+  // Bar chart
+  new Chart(ctx, {
+    type: "horizontalBar",
+    data: {
+      labels: labels,
+      datasets: [
+        {
+          data: chartData,
+          backgroundColor: [
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/content/barchart/script.js:62]: Background colors can also come from the JSON file.

File: application/backend/src/main/resources/apps/typerefinery/components/content/barchart/script.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+              },
+            },
+            scaleLabel: {
+              display: true,
+              padding: 10,
+              fontFamily: "Lato",
+              fontColor: "#555759",
+              fontSize: 16,
+              fontStyle: 700,
+              labelString: "Scale Label",
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/content/barchart/script.js:131]: Label should also come from JSON

File: application/backend/src/main/resources/apps/typerefinery/components/content/barchart/template.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<template data-sly-template.barchart="${ @ model }">
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/content/barchart/template.html:1]: camel case convention should be used here.

barchart to barChart

**Review by Harikarthyk — CHANGES_REQUESTED** [review]

File: application/backend/src/main/resources/apps/typerefinery/components/content/barchart/barchart.html (hunk 2 above)
Comment (rohitgupta274509) [reply application/backend/src/main/resources/apps/typerefinery/components/content/barchart/barchart.html]: change completed.

File: application/backend/src/main/resources/apps/typerefinery/components/content/barchart/script.js (hunk 3 above)
Comment (rohitgupta274509) [reply application/backend/src/main/resources/apps/typerefinery/components/content/barchart/script.js]: done

File: application/backend/src/main/resources/apps/typerefinery/components/content/barchart/script.js (hunk 4 above)
Comment (rohitgupta274509) [reply application/backend/src/main/resources/apps/typerefinery/components/content/barchart/script.js]: added in json

File: application/backend/src/main/resources/apps/typerefinery/components/content/barchart/template.html (hunk 5 above)
Comment (rohitgupta274509) [reply application/backend/src/main/resources/apps/typerefinery/components/content/barchart/template.html]: done

**Review by Harikarthyk — APPROVED** [review]

**Pull request merged**